src/core/public/saved_objects/saved_objects_client.ts

```diff
@@ -226,7 +220,9 @@ export class SavedObjectsClient {
       .then(resp => this.createSavedObject(resp))
       .catch((error: object) => {
         if (isAutoCreateIndexError(error)) {
-          showAutoCreateIndexErrorPage();
+          window.location.assign(
+            this.http.basePath.prepend('/app/kibana#/error/action.auto_create_index')
+          );
```

Comment:
TODO: test this case

Reply:
🙈 on this `if (isAutoCreateIndexError(error)) {` block, but we need to keep the `TODO` to at least adapt the redirecting url when the page is migrated from legacy.

Reply:
is there a test for this?

Reply:
We can remove the auto create index error handling code and views. Probably out of scope for this PR, so I created https://github.com/elastic/kibana/issues/63894